### ISSUE TITLE
Cleanup naming of internal functions?

### ISSUE THREAD

**Issue by minad** [libtom/libtommath#164]
~~~ c
int s_mp_add(const mp_int *a, const mp_int *b, mp_int *c);
int s_mp_sub(const mp_int *a, const mp_int *b, mp_int *c);
#define s_mp_mul(a, b, c) s_mp_mul_digs(a, b, c, (a)->used + (b)->used + 1)
int fast_s_mp_mul_digs(const mp_int *a, const mp_int *b, mp_int *c, int digs);
int s_mp_mul_digs(const mp_int *a, const mp_int *b, mp_int *c, int digs);
int fast_s_mp_mul_high_digs(const mp_int *a, const mp_int *b, mp_int *c, int digs);
int s_mp_mul_high_digs(const mp_int *a, const mp_int *b, mp_int *c, int digs);
int fast_s_mp_sqr(const mp_int *a, mp_int *b);
int s_mp_sqr(const mp_int *a, mp_int *b);
int mp_karatsuba_mul(const mp_int *a, const mp_int *b, mp_int *c);
int mp_toom_mul(const mp_int *a, const mp_int *b, mp_int *c);
int mp_karatsuba_sqr(const mp_int *a, mp_int *b);
int mp_toom_sqr(const mp_int *a, mp_int *b);
int fast_mp_invmod(const mp_int *a, const mp_int *b, mp_int *c);
int mp_invmod_slow(const mp_int *a, const mp_int *b, mp_int *c);
int fast_mp_montgomery_reduce(mp_int *x, const mp_int *n, mp_digit rho);
int mp_exptmod_fast(const mp_int *G, const mp_int *X, const mp_int *P, mp_int *Y, int redmode);
int s_mp_exptmod(const mp_int *G, const mp_int *X, const mp_int *P, mp_int *Y, int redmode);
void bn_reverse(unsigned char *s, int len);

extern const char *const mp_s_rmap;
extern const uint8_t mp_s_rmap_reverse[];
extern const size_t mp_s_rmap_reverse_sz;
~~~

`s_mp_*` for static (or private nonstatic) functions is not used consistently

**Comment by czurnieden** [comment]
It doesn't look nice but it works for everybody and isn't very painful for us to work with, he says, fully aware of how that sounds like. ;-)

I wouldn't touch the API in LTM, you would inconvenience some users, that is: break their stuff. Such things need to get announced early enough and loud enough and at least two releases in advance.

Otherwise: yes, why not.

**Comment by minad** [comment]
I wouldn't touch the public API. Just talking about private stuff. There is no reason to deprecate anything. However this issue is also not very important and does not really need "fixing".

What I consider more important is to create a proper test suite.

**Comment by sjaeckel** [comment]
> I wouldn't touch the public API. Just talking about private stuff.

There's one real problem here, we have two separate headers, a public and a private one, that's true, but since none of the symbols in the private header is properly hidden, they're accessible and from a `ld` PoV they're public... That's also the reason why I've still not merged #155 as it would create an incompatible API & ABI for x32 only and I'm not sure if we really want that...

(P.S. I had a much longer comment on this, but then somehow github crashed a tab in my browser and now it's that short and I'm tired of re-writing everything...)

**Comment by sjaeckel** [comment]
> incompatible API & ABI ... and I'm not sure if we really want that

Which doesn't mean that it has to stay like that forever, but we should coordinate such a change as already indicated by @czurnieden

**Comment by minad** [comment]
closed via #172. discussion can continue there.
